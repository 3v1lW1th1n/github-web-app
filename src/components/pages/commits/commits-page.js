import React, { useState, useEffect } from 'react'
import styled from 'styled-components'
import { toastr } from 'react-redux-toastr'
import Layout from '../../layout'
import Loader from '../../loader'
import Pagination from '../../pagination'
import SearchInput from '../../search-input'
import CommitsList from '../../commits-list'

const CommitsPage = ({
  className,
  isLoading,
  getCommits,
  commits,
  errorMessage,
  match,
  location,
  history
}) => {
  const [filteredCommits, setFilteredCommits] = useState([])
  const [pageItems, setPageItems] = useState([])
  const [filter, setFilter] = useState('')

  const getCurrentPage = () => {
    const search = new URLSearchParams(location.search)
    const currentPage = parseInt(search.get('page'))
    return currentPage || 1
  }

  const filterCommits = () => {
    let filteredCommits = commits.nodes
    if (filter) {
      filteredCommits = filteredCommits.filter(
        ({ commit }) => commit.message.toLowerCase().includes(filter)
      )
    }
    setFilteredCommits(filteredCommits)
  }

  const handleChangePage = (pageItems, page) => {
    setPageItems(pageItems)
    history.push(`?page=${page}`)
  }

  const handleFilter = query => setFilter(query)

  useEffect(() => {
    const { user, repo } = match.params
    getCommits && getCommits(user, repo)
  }, [])

  useEffect(() => filterCommits(), [commits, filter])

  useEffect(() => {
    errorMessage && toastr.error(errorMessage)
  }, [errorMessage])

  return (
    <Layout className={className}>
      <SearchInput
        className='commits-page__search-input'
        placeholder='pesquisar...'
        onRequest={handleFilter}
      />
      {isLoading
        ? <Loader />
        : <div>
          <CommitsList commits={pageItems} />
          <Pagination
            items={filteredCommits}
            initialPage={getCurrentPage()}
            onChangePage={handleChangePage}
          />
        </div>
      }
    </Layout>
  )
}

const StyledCommitsPage = styled(CommitsPage)`
  .commits-page__search-input {
    margin: 32px auto;
  }
`

StyledCommitsPage.displayName = 'CommitsPage'

StyledCommitsPage.defaultProps = {
  commits: []
}

export default StyledCommitsPage
