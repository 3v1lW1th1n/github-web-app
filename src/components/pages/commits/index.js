export { default } from './commits-page'
