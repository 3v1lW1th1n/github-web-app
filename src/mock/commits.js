export default [
  {
    sha: '6c13ea200d12191fbfe6acab62a2eaaab5de96dd',
    node_id: 'MDY6Q29tbWl0MTYyOTMyNTY3OjZjMTNlYTIwMGQxMjE5MWZiZmU2YWNhYjYyYTJlYWFhYjVkZTk2ZGQ=',
    commit: {
      author: {
        name: 'Al Piepho',
        email: 'alpiepho@yahoo.com',
        date: '2019-06-30T15:59:53Z'
      },
      committer: {
        name: 'Francisco Santos',
        email: 'yoshi.df@gmail.com',
        date: '2019-06-30T15:59:53Z'
      },
      message: 'Adding check for a default Google Analytics ID (#11)',
      tree: {
        sha: 'c5cd024519050877ec312e24238ff31ba968b0a3',
        url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/git/trees/c5cd024519050877ec312e24238ff31ba968b0a3'
      },
      url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/git/commits/6c13ea200d12191fbfe6acab62a2eaaab5de96dd',
      comment_count: 0,
      verification: {
        verified: false,
        reason: 'unsigned',
        signature: null,
        payload: null
      }
    },
    url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/6c13ea200d12191fbfe6acab62a2eaaab5de96dd',
    html_url: 'https://github.com/santosfrancisco/gatsby-starter-cv/commit/6c13ea200d12191fbfe6acab62a2eaaab5de96dd',
    comments_url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/6c13ea200d12191fbfe6acab62a2eaaab5de96dd/comments',
    author: {
      login: 'alpiepho',
      id: 8628212,
      node_id: 'MDQ6VXNlcjg2MjgyMTI=',
      avatar_url: 'https://avatars2.githubusercontent.com/u/8628212?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/alpiepho',
      html_url: 'https://github.com/alpiepho',
      followers_url: 'https://api.github.com/users/alpiepho/followers',
      following_url: 'https://api.github.com/users/alpiepho/following{/other_user}',
      gists_url: 'https://api.github.com/users/alpiepho/gists{/gist_id}',
      starred_url: 'https://api.github.com/users/alpiepho/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/alpiepho/subscriptions',
      organizations_url: 'https://api.github.com/users/alpiepho/orgs',
      repos_url: 'https://api.github.com/users/alpiepho/repos',
      events_url: 'https://api.github.com/users/alpiepho/events{/privacy}',
      received_events_url: 'https://api.github.com/users/alpiepho/received_events',
      type: 'User',
      site_admin: false
    },
    committer: {
      login: 'santosfrancisco',
      id: 15852005,
      node_id: 'MDQ6VXNlcjE1ODUyMDA1',
      avatar_url: 'https://avatars3.githubusercontent.com/u/15852005?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/santosfrancisco',
      html_url: 'https://github.com/santosfrancisco',
      followers_url: 'https://api.github.com/users/santosfrancisco/followers',
      following_url: 'https://api.github.com/users/santosfrancisco/following{/other_user}',
      gists_url: 'https://api.github.com/users/santosfrancisco/gists{/gist_id}',
      starred_url: 'https://api.github.com/users/santosfrancisco/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/santosfrancisco/subscriptions',
      organizations_url: 'https://api.github.com/users/santosfrancisco/orgs',
      repos_url: 'https://api.github.com/users/santosfrancisco/repos',
      events_url: 'https://api.github.com/users/santosfrancisco/events{/privacy}',
      received_events_url: 'https://api.github.com/users/santosfrancisco/received_events',
      type: 'User',
      site_admin: false
    },
    parents: [
      {
        sha: 'a18d79eb4d6c582642d7195f588abc3269243e13',
        url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/a18d79eb4d6c582642d7195f588abc3269243e13',
        html_url: 'https://github.com/santosfrancisco/gatsby-starter-cv/commit/a18d79eb4d6c582642d7195f588abc3269243e13'
      }
    ]
  },
  {
    sha: 'a18d79eb4d6c582642d7195f588abc3269243e13',
    node_id: 'MDY6Q29tbWl0MTYyOTMyNTY3OmExOGQ3OWViNGQ2YzU4MjY0MmQ3MTk1ZjU4OGFiYzMyNjkyNDNlMTM=',
    commit: {
      author: {
        name: 'renovate[bot]',
        email: 'renovate[bot]@users.noreply.github.com',
        date: '2019-05-04T22:03:51Z'
      },
      committer: {
        name: 'Francisco Santos',
        email: 'yoshi.df@gmail.com',
        date: '2019-05-04T22:03:51Z'
      },
      message: 'Pin dependencies (#4)',
      tree: {
        sha: '0ec843db714cb3cfeed167ae43cd2ab857e11aca',
        url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/git/trees/0ec843db714cb3cfeed167ae43cd2ab857e11aca'
      },
      url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/git/commits/a18d79eb4d6c582642d7195f588abc3269243e13',
      comment_count: 0,
      verification: {
        verified: false,
        reason: 'unsigned',
        signature: null,
        payload: null
      }
    },
    url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/a18d79eb4d6c582642d7195f588abc3269243e13',
    html_url: 'https://github.com/santosfrancisco/gatsby-starter-cv/commit/a18d79eb4d6c582642d7195f588abc3269243e13',
    comments_url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/a18d79eb4d6c582642d7195f588abc3269243e13/comments',
    author: {
      login: 'renovate[bot]',
      id: 29139614,
      node_id: 'MDM6Qm90MjkxMzk2MTQ=',
      avatar_url: 'https://avatars1.githubusercontent.com/in/2740?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/renovate%5Bbot%5D',
      html_url: 'https://github.com/apps/renovate',
      followers_url: 'https://api.github.com/users/renovate%5Bbot%5D/followers',
      following_url: 'https://api.github.com/users/renovate%5Bbot%5D/following{/other_user}',
      gists_url: 'https://api.github.com/users/renovate%5Bbot%5D/gists{/gist_id}',
      starred_url: 'https://api.github.com/users/renovate%5Bbot%5D/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/renovate%5Bbot%5D/subscriptions',
      organizations_url: 'https://api.github.com/users/renovate%5Bbot%5D/orgs',
      repos_url: 'https://api.github.com/users/renovate%5Bbot%5D/repos',
      events_url: 'https://api.github.com/users/renovate%5Bbot%5D/events{/privacy}',
      received_events_url: 'https://api.github.com/users/renovate%5Bbot%5D/received_events',
      type: 'Bot',
      site_admin: false
    },
    committer: {
      login: 'santosfrancisco',
      id: 15852005,
      node_id: 'MDQ6VXNlcjE1ODUyMDA1',
      avatar_url: 'https://avatars3.githubusercontent.com/u/15852005?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/santosfrancisco',
      html_url: 'https://github.com/santosfrancisco',
      followers_url: 'https://api.github.com/users/santosfrancisco/followers',
      following_url: 'https://api.github.com/users/santosfrancisco/following{/other_user}',
      gists_url: 'https://api.github.com/users/santosfrancisco/gists{/gist_id}',
      starred_url: 'https://api.github.com/users/santosfrancisco/starred{/owner}{/repo}',
      subscriptions_url: 'https://api.github.com/users/santosfrancisco/subscriptions',
      organizations_url: 'https://api.github.com/users/santosfrancisco/orgs',
      repos_url: 'https://api.github.com/users/santosfrancisco/repos',
      events_url: 'https://api.github.com/users/santosfrancisco/events{/privacy}',
      received_events_url: 'https://api.github.com/users/santosfrancisco/received_events',
      type: 'User',
      site_admin: false
    },
    parents: [
      {
        sha: 'f007950d6d6338bc30d59b750de2d0e0b2dbbaef',
        url: 'https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/f007950d6d6338bc30d59b750de2d0e0b2dbbaef',
        html_url: 'https://github.com/santosfrancisco/gatsby-starter-cv/commit/f007950d6d6338bc30d59b750de2d0e0b2dbbaef'
      }
    ]
  }
]
