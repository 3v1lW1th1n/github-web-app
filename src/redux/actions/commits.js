import { gql } from '../api'
import * as constants from '../constants'

export const getCommits = (user, repo) => {
  const getCommits = `{ 
      repository(name: "${repo}", owner: "${user}") {
        ref(qualifiedName: "master") {
          target {
            ...on Commit {
              history(first: 10) {
                totalCount
                nodes {
                  oid
                  messageHeadline
                  url
                  author {
                    name
                  }
                  committedDate
                }
                pageInfo {
                  hasNextPage
                  hasPreviousPage
                }
              }
            }
          }
        }
      }
    }
  `
  return dispatch => {
    dispatch({ type: constants.GET_COMMITS_REQUEST })
    return gql.post('', { query: getCommits })
      .then(({ data: { data: { repository: { ref: { target: { history } } } } } }) => {
        return dispatch({ type: constants.GET_COMMITS_SUCCESS, payload: history })
      })
      .catch(error => dispatch({ type: constants.GET_COMMITS_FAILURE, payload: error.message }))
  }
}
