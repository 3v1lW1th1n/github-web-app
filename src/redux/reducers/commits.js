import {
  GET_COMMITS_SUCCESS
} from '../constants'

export const commits = (state = [], action) => {
  switch (action.type) {
    case GET_COMMITS_SUCCESS:
      return action.payload
    default:
      return state
  }
}
