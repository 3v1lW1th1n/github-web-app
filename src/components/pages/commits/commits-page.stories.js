import React from 'react'
import { storiesOf } from '@storybook/react'
import { MemoryRouter, Route } from 'react-router-dom'
import CommitsPage from './'

import commits from '../../../mock/commits'

storiesOf('Pages/CommitsPage', module)
  .add('default', () => (
    <MemoryRouter initialEntries={['/']}>
      <Route path='/' component={props => <CommitsPage commits={commits} {...props} />} />
    </MemoryRouter>
  )
  )
