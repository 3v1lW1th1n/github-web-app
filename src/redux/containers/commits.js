import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getCommits } from '../actions/commits'
import { getIsLoading } from '../selectors/util'
import Commits from '../../components/pages/commits'

const CommitsContainer = props => <Commits {...props} />

const mapStateToProps = state => ({
  commits: state.commits,
  isLoading: getIsLoading(state),
  errorMessage: state.errorMessage
})

const mapDispatchToProps = dispatch =>
  bindActionCreators({ getCommits }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(CommitsContainer)
