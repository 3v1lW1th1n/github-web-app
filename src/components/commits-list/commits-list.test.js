import 'jest-styled-components'

import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import CommitsList from './'

import commits from '../../mock/commits'

describe('<CommitsList />', () => {
  it('should match default snapshot', () => {
    const component = shallow(<CommitsList commits={commits} />)
    expect(component).toMatchSnapshot()
  })

  it('should render the commits correctly', () => {
    const component = mount(
      <MemoryRouter>
        <CommitsList commits={commits} />
      </MemoryRouter>)

    expect(component.find('li').length).toEqual(commits.length)
  })

  it('should render the empty list message if no commits exists', () => {
    const component = mount(
      <MemoryRouter>
        <CommitsList commits={[]} />
      </MemoryRouter>)

    expect(component.find('EmptyList').exists()).toBe(true)
  })

  it('should have the right displayName', () => {
    expect(CommitsList.displayName).toEqual('CommitsList')
  })
})
