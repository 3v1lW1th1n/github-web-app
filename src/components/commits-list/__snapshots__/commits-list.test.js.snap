// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommitsList /> should match default snapshot 1`] = `
<StyledComponent
  commits={
    Array [
      Object {
        "author": Object {
          "avatar_url": "https://avatars2.githubusercontent.com/u/8628212?v=4",
          "events_url": "https://api.github.com/users/alpiepho/events{/privacy}",
          "followers_url": "https://api.github.com/users/alpiepho/followers",
          "following_url": "https://api.github.com/users/alpiepho/following{/other_user}",
          "gists_url": "https://api.github.com/users/alpiepho/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/alpiepho",
          "id": 8628212,
          "login": "alpiepho",
          "node_id": "MDQ6VXNlcjg2MjgyMTI=",
          "organizations_url": "https://api.github.com/users/alpiepho/orgs",
          "received_events_url": "https://api.github.com/users/alpiepho/received_events",
          "repos_url": "https://api.github.com/users/alpiepho/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/alpiepho/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/alpiepho/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/alpiepho",
        },
        "comments_url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/6c13ea200d12191fbfe6acab62a2eaaab5de96dd/comments",
        "commit": Object {
          "author": Object {
            "date": "2019-06-30T15:59:53Z",
            "email": "alpiepho@yahoo.com",
            "name": "Al Piepho",
          },
          "comment_count": 0,
          "committer": Object {
            "date": "2019-06-30T15:59:53Z",
            "email": "yoshi.df@gmail.com",
            "name": "Francisco Santos",
          },
          "message": "Adding check for a default Google Analytics ID (#11)",
          "tree": Object {
            "sha": "c5cd024519050877ec312e24238ff31ba968b0a3",
            "url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/git/trees/c5cd024519050877ec312e24238ff31ba968b0a3",
          },
          "url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/git/commits/6c13ea200d12191fbfe6acab62a2eaaab5de96dd",
          "verification": Object {
            "payload": null,
            "reason": "unsigned",
            "signature": null,
            "verified": false,
          },
        },
        "committer": Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/15852005?v=4",
          "events_url": "https://api.github.com/users/santosfrancisco/events{/privacy}",
          "followers_url": "https://api.github.com/users/santosfrancisco/followers",
          "following_url": "https://api.github.com/users/santosfrancisco/following{/other_user}",
          "gists_url": "https://api.github.com/users/santosfrancisco/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/santosfrancisco",
          "id": 15852005,
          "login": "santosfrancisco",
          "node_id": "MDQ6VXNlcjE1ODUyMDA1",
          "organizations_url": "https://api.github.com/users/santosfrancisco/orgs",
          "received_events_url": "https://api.github.com/users/santosfrancisco/received_events",
          "repos_url": "https://api.github.com/users/santosfrancisco/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/santosfrancisco/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/santosfrancisco/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/santosfrancisco",
        },
        "html_url": "https://github.com/santosfrancisco/gatsby-starter-cv/commit/6c13ea200d12191fbfe6acab62a2eaaab5de96dd",
        "node_id": "MDY6Q29tbWl0MTYyOTMyNTY3OjZjMTNlYTIwMGQxMjE5MWZiZmU2YWNhYjYyYTJlYWFhYjVkZTk2ZGQ=",
        "parents": Array [
          Object {
            "html_url": "https://github.com/santosfrancisco/gatsby-starter-cv/commit/a18d79eb4d6c582642d7195f588abc3269243e13",
            "sha": "a18d79eb4d6c582642d7195f588abc3269243e13",
            "url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/a18d79eb4d6c582642d7195f588abc3269243e13",
          },
        ],
        "sha": "6c13ea200d12191fbfe6acab62a2eaaab5de96dd",
        "url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/6c13ea200d12191fbfe6acab62a2eaaab5de96dd",
      },
      Object {
        "author": Object {
          "avatar_url": "https://avatars1.githubusercontent.com/in/2740?v=4",
          "events_url": "https://api.github.com/users/renovate%5Bbot%5D/events{/privacy}",
          "followers_url": "https://api.github.com/users/renovate%5Bbot%5D/followers",
          "following_url": "https://api.github.com/users/renovate%5Bbot%5D/following{/other_user}",
          "gists_url": "https://api.github.com/users/renovate%5Bbot%5D/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/apps/renovate",
          "id": 29139614,
          "login": "renovate[bot]",
          "node_id": "MDM6Qm90MjkxMzk2MTQ=",
          "organizations_url": "https://api.github.com/users/renovate%5Bbot%5D/orgs",
          "received_events_url": "https://api.github.com/users/renovate%5Bbot%5D/received_events",
          "repos_url": "https://api.github.com/users/renovate%5Bbot%5D/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/renovate%5Bbot%5D/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/renovate%5Bbot%5D/subscriptions",
          "type": "Bot",
          "url": "https://api.github.com/users/renovate%5Bbot%5D",
        },
        "comments_url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/a18d79eb4d6c582642d7195f588abc3269243e13/comments",
        "commit": Object {
          "author": Object {
            "date": "2019-05-04T22:03:51Z",
            "email": "renovate[bot]@users.noreply.github.com",
            "name": "renovate[bot]",
          },
          "comment_count": 0,
          "committer": Object {
            "date": "2019-05-04T22:03:51Z",
            "email": "yoshi.df@gmail.com",
            "name": "Francisco Santos",
          },
          "message": "Pin dependencies (#4)",
          "tree": Object {
            "sha": "0ec843db714cb3cfeed167ae43cd2ab857e11aca",
            "url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/git/trees/0ec843db714cb3cfeed167ae43cd2ab857e11aca",
          },
          "url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/git/commits/a18d79eb4d6c582642d7195f588abc3269243e13",
          "verification": Object {
            "payload": null,
            "reason": "unsigned",
            "signature": null,
            "verified": false,
          },
        },
        "committer": Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/15852005?v=4",
          "events_url": "https://api.github.com/users/santosfrancisco/events{/privacy}",
          "followers_url": "https://api.github.com/users/santosfrancisco/followers",
          "following_url": "https://api.github.com/users/santosfrancisco/following{/other_user}",
          "gists_url": "https://api.github.com/users/santosfrancisco/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/santosfrancisco",
          "id": 15852005,
          "login": "santosfrancisco",
          "node_id": "MDQ6VXNlcjE1ODUyMDA1",
          "organizations_url": "https://api.github.com/users/santosfrancisco/orgs",
          "received_events_url": "https://api.github.com/users/santosfrancisco/received_events",
          "repos_url": "https://api.github.com/users/santosfrancisco/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/santosfrancisco/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/santosfrancisco/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/santosfrancisco",
        },
        "html_url": "https://github.com/santosfrancisco/gatsby-starter-cv/commit/a18d79eb4d6c582642d7195f588abc3269243e13",
        "node_id": "MDY6Q29tbWl0MTYyOTMyNTY3OmExOGQ3OWViNGQ2YzU4MjY0MmQ3MTk1ZjU4OGFiYzMyNjkyNDNlMTM=",
        "parents": Array [
          Object {
            "html_url": "https://github.com/santosfrancisco/gatsby-starter-cv/commit/f007950d6d6338bc30d59b750de2d0e0b2dbbaef",
            "sha": "f007950d6d6338bc30d59b750de2d0e0b2dbbaef",
            "url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/f007950d6d6338bc30d59b750de2d0e0b2dbbaef",
          },
        ],
        "sha": "a18d79eb4d6c582642d7195f588abc3269243e13",
        "url": "https://api.github.com/repos/santosfrancisco/gatsby-starter-cv/commits/a18d79eb4d6c582642d7195f588abc3269243e13",
      },
    ]
  }
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "commits": Array [],
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "StyledCommitsList-sc-1njrout-0",
        "isStatic": true,
        "rules": Array [
          ".commits-list__title{display:flex;align-items:center;}.commits-list__link{display:flex;text-decoration:none;color:#000;font-weight:bold;}.commits-list__separator{border:none;border-bottom:1px solid rgba(0,0,0,.2);}",
        ],
      },
      "displayName": "CommitsList",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "StyledCommitsList-sc-1njrout-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
